### ISSUE TITLE
yaml syntax broken

### ISSUE THREAD

**Issue by eregnier** [rochacbruno/flasgger#181]
Hello
I have an issue now that is 
https://gist.github.com/anonymous/904c4628fc3f9d23870a915cf0111610

because we have some files like
https://zerobin.net/?e3390f9a981a3ccd#mAvv8zX8ItzuQPs6qxIqXVPDc03d4yspKeQ1mQTeYms=

and our flasgger version is
flasgger==0.8.0
previously 
flasgger==0.6.3

It was working fine until I upgraded flasgger version. (maybe I done something else, but I don't mind what)

Any idea on if I should do something or if flasgger should change something ?

**Comment by eregnier** [comment]
Edit : I just tried to run swagger with flasgger 0.8.0 instead of 0.8.1 and it works (note that in my initial issue there is a mistake : I meant we upgraded to 0.8.1 where the breaking change seems to appear.

**Comment by rochacbruno** [comment]
```
---
Get a single conversation from id.

The state field in conversation payload may take one of the following status

  - **normalState** : Default state for conversation

  - **waitForFirstMessage** : Message count equals 0

  - **waitingForPermission** : Message count equals 1 and current user created conversation

  - **receivedInvitation** : Message count equals 1 and current user DID NOT created conversation

  - **iAmBlocking** : Message count superior to 1 and current user blocked conversation

  - **iAmBlocked** : Message count superior to 1 and conversation other user blocked conversation

  - **weAreBothBlocked** : Message count superior to 1 and both users blocked conversation
---
tags:
  - Chat / Conversations
security:
  - APIKeyQueryParam: []
  - APIKeyHeader: []
parameters:
  - name: conversation_id
    in:  path
    required: true
    schema:
        type: integer
    description: The convesation to fetch.
responses:
  404:
    description: Conversation not found in database.
    schema:
      $ref: '#/definitions/NotFound'
  200:
    description: Return convesation dumped object.
    schema:
      $ref: '#/definitions/Conversation'
  401:
    description: Not authorized, user must be logged to perform query
    schema:
      $ref: '#/definitions/Unauthorized'

```

**Comment by rochacbruno** [comment]
```
Traceback (most recent call last):
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/werkzeug/contrib/fixers.py", line 152, in __call__
    return self.app(environ, start_response)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask_debugtoolbar/__init__.py", line 125, in dispatch_request
    return view_func(**req.view_args)
  File "/usr/lib/python3.5/cProfile.py", line 109, in runcall
    return func(*args, **kw)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flask/views.py", line 149, in dispatch_request
    return meth(*args, **kwargs)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flasgger/base.py", line 189, in get
    optional_fields, self.process_doc)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flasgger/utils.py", line 131, in get_specs
    method, sanitizer, endpoint=rule.endpoint, verb=verb)
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/flasgger/utils.py", line 527, in parse_docstring
    swag = yaml.load(full_doc[yaml_sep + 4:])
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/yaml/__init__.py", line 72, in load
    return loader.get_single_data()
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/yaml/constructor.py", line 35, in get_single_data
    node = self.get_single_node()
  File "/home/utopman/.local/share/virtualenvs/jaccede-api-v4-w1kPGGna/lib/python3.5/site-packages/yaml/composer.py", line 43, in get_single_node
    event.start_mark)
yaml.composer.ComposerError: expected a single document in the stream
  in "<unicode string>", line 1, column 1:
    Perform search on places in Jacc ... 
    ^
but found another document
  in "<unicode string>", line 4, column 1:
    ---

```

**Comment by rochacbruno** [comment]
@eregnier I'll investigate the issue

**Comment by eregnier** [comment]
I missed to share an information : When I tried to investigate a bit on this issue I found this thread which is related in a sense, but with no more troubleshooting / experiments I am not really sure. Here is the thread I am speaking about
https://stackoverflow.com/questions/14359557/reading-yaml-in-python

**Comment by javabrett** [comment]
My understanding is that _Flasgger_ is tolerant of non-YAML content in a pre-document directives block, but that it still expects a single document, and that is inline with this error.  The exception could probably be caught and some additional help could be provided.

So _Flasgger_ will tolerate:

```
Some documentation for the API.
It can even contain bare colons : like that.
Here we are essentially in the YAML directives block of document 1.
Next follows three dashes then the spec YAML.
---
# this is the actual Swagger spec
tags:
  - Chat / Conversations
```

The problem document here starts with `---`, which means "start of YAML document with no directives block".  Then follows text-not-YAML.  Following that is a valid, second YAML document.  Without the first `---` it should work fine.

TL;DR _Flasgger_ needs a single YAML document, above two are present.

Is there a reason for the first `---` - could it be removed?

**Comment by eregnier** [comment]
We fixed the file on the project. I do not have this problem anymore. The problem may persist for some other people that have formatted their documentation like me, but this thread does not move a lot. Then I suppose there is not much problem remaining at the moment. 

You may close the issue. 

Thank you.
